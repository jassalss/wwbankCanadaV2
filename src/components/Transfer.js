import React, { Component } from "react";
import { connect } from "react-redux";
import { transferTheMoney } from "../actions";
import Converter from "../Converter";
class Transfer extends Component {
  handleSubmit = (event) => {
    event.preventDefault();
    let fromAccount = event.target.fromaccountNumber.value;
    let toAccount = event.target.toaccountNumber.value;
    let trasnferAmount = event.target.amount.value;
    let currency = event.target.currency.value;
    let CustomerName = event.target.customerName.value;
    let CustomerId = event.target.customerId.value;
    let otherCurrency = "";
    if (trasnferAmount < 0) {
      alert("Transfer amount should more than 0");
      return;
    }
    if (currency !== "CAD") {
      otherCurrency = ` = $${trasnferAmount} ${currency}`;
      trasnferAmount = Converter(trasnferAmount, currency);
    }
    this.props.transferTheMoney(
      toAccount,
      fromAccount,
      trasnferAmount,
      otherCurrency,
      CustomerName,
      CustomerId
    );
  };
  render() {
    if (this.props.trasferInfo) {
      alert(this.props.trasferInfo);
      window.location.reload();
    }
    return (
      <div>
        <h3 className="ui center aligned header">Transfer Money</h3>
        <div className="ui centered grid">
          <div className="ten wide column">
            <form className="ui form" onSubmit={this.handleSubmit}>
              <div className="field">
                <select className="ui dropdown" required name="currency">
                  <option value="">Select Currency</option>
                  <option value="CAD">CAD</option>
                  <option value="USD">USD</option>
                  <option value="MXN">MXN</option>
                </select>
              </div>
              <div className="field">
                <label>Trasfer Amount</label>
                <input
                  type="number"
                  name="amount"
                  step="0.001"
                  placeholder="Amount for transfer"
                />
              </div>
              <div className="field">
                <label>From Account</label>
                <input
                  required
                  type="number"
                  name="fromaccountNumber"
                  placeholder="From Account"
                />
              </div>
              <div className="field">
                <label>To Account</label>
                <input
                  required
                  type="number"
                  name="toaccountNumber"
                  placeholder="To Account"
                />
              </div>
              <div className="field">
                <label>Customer Name</label>
                <input
                  required
                  type="text"
                  name="customerName"
                  placeholder="Customer Name"
                />
              </div>
              <div className="field">
                <label>Customer ID</label>
                <input
                  required
                  type="number"
                  name="customerId"
                  placeholder="Customer ID (Input is Number not letter)"
                />
              </div>
              <button className="ui button" type="submit">
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    trasferInfo: state.bankAccountsInfo.trasferInfo,
  };
};
export default connect(mapStateToProps, { transferTheMoney })(Transfer);
