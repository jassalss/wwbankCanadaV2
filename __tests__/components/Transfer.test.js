import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import "babel-polyfill";
// Component to be tested
import Transfer from "../../src/components/Transfer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureStore([]);

let bankAccountsInfo = {
  savingMessage: "test",
  depositInfo: "test",
  withDrawInfo: "test",
  trasferInfo: "test",
};
describe("Testing Transfer page with Snapshot", () => {
  let store;
  window.alert = jest.fn();
  beforeEach(() => {
    store = mockStore({
      bankAccountsInfo,
    });
  });

  it("matches the snapshot", () => {
    const { location } = window;
    delete window.location;
    window.location = { reload: jest.fn() };
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Transfer />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    window.location = location;
  });
  window.alert.mockClear();
});
