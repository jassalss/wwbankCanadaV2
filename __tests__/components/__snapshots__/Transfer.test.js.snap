// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Transfer page with Snapshot matches the snapshot 1`] = `
<div>
  <h3
    className="ui center aligned header"
  >
    Transfer Money
  </h3>
  <div
    className="ui centered grid"
  >
    <div
      className="ten wide column"
    >
      <form
        className="ui form"
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <select
            className="ui dropdown"
            name="currency"
            required={true}
          >
            <option
              value=""
            >
              Select Currency
            </option>
            <option
              value="CAD"
            >
              CAD
            </option>
            <option
              value="USD"
            >
              USD
            </option>
            <option
              value="MXN"
            >
              MXN
            </option>
          </select>
        </div>
        <div
          className="field"
        >
          <label>
            Trasfer Amount
          </label>
          <input
            name="amount"
            placeholder="Amount for transfer"
            step="0.001"
            type="number"
          />
        </div>
        <div
          className="field"
        >
          <label>
            From Account
          </label>
          <input
            name="fromaccountNumber"
            placeholder="From Account"
            required={true}
            type="number"
          />
        </div>
        <div
          className="field"
        >
          <label>
            To Account
          </label>
          <input
            name="toaccountNumber"
            placeholder="To Account"
            required={true}
            type="number"
          />
        </div>
        <div
          className="field"
        >
          <label>
            Customer Name
          </label>
          <input
            name="customerName"
            placeholder="Customer Name"
            required={true}
            type="text"
          />
        </div>
        <div
          className="field"
        >
          <label>
            Customer ID
          </label>
          <input
            name="customerId"
            placeholder="Customer ID (Input is Number not letter)"
            required={true}
            type="number"
          />
        </div>
        <button
          className="ui button"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
